modules/custom/az_core/src/AZContentFieldUpdater.php

```diff
@@ -0,0 +1,326 @@
+            foreach ($field_item->getProperties() as $property => $value) {
+              $new_value[$property] = ($property === $options['value_key']) ? $processed_value : $value;
+            }
+            $field->get($delta)->setValue($new_value);
+            $needs_update = TRUE;
+          }
+        }
+      }
+
+      if ($needs_update) {
+        // Create new revision if requested.
+        if ($options['create_revisions'] && $entity instanceof RevisionableInterface) {
+          $entity->setNewRevision(TRUE);
+          $entity->isDefaultRevision(TRUE);
+
+          // Add revision metadata for non-paragraph entities.
+          if ($entity instanceof RevisionLogInterface && !$entity instanceof ParagraphInterface) {
+            $time = \Drupal::time()->getRequestTime();
+            $entity->setRevisionCreationTime($time);
+            $entity->setRevisionUserId(1);
+            $message_args = [
+              '@field' => $field_name,
+              '@type' => $entity->getEntityTypeId(),
+              '@id' => $entity->id(),
+            ];
+            if (!empty($options['bundle_name'])) {
+              $message_args['@bundle'] = $options['bundle_name'];
+              $message = $this->t('Updated field @field on @bundle @type @id', $message_args);
+            }
+            else {
+              $message = $this->t('Updated field @field on @type @id', $message_args);
+            }
+            // Format and add prefix/suffix to revision log message.
+            if ($options['prefix']) {
+              $message = $options['prefix'] . ': ' . $message;
+            }
+            if ($options['suffix']) {
+              $message .= ' (' . $options['suffix'] . ')';
+            }
+            $entity->setRevisionLogMessage($message);
+          }
+        }
+
+        // Save the entity.
+        $entity->save();
+
+        // Handle parent entity revisions for paragraphs.
+        if ($options['create_revisions'] &&
+            $entity instanceof ParagraphInterface &&
+            $parent = $entity->getParentEntity()) {
+          if ($parent instanceof RevisionableInterface) {
+            // Load fresh version of parent.
+            $parent_storage = $this->entityTypeManager->getStorage($parent->getEntityTypeId());
+            /** @var \Drupal\Core\Entity\ContentEntityInterface $parent */
+            $parent = $parent_storage->load($parent->id());
+
+            // Create new revision.
+            $parent->setNewRevision(TRUE);
+            $parent->isDefaultRevision(TRUE);
+
+            // Update revision metadata.
+            if ($parent instanceof RevisionLogInterface && !$parent instanceof ParagraphInterface) {
+              $time = \Drupal::time()->getRequestTime();
+              $parent->setRevisionCreationTime($time);
+              $parent->setRevisionUserId(1);
+              if ($parent instanceof TranslatableRevisionableInterface) {
+                $parent->setRevisionTranslationAffected(TRUE);
```

Comment:
This doesn't need to block merging, but I'd like to create a followup issue that blocks the stable release to determine why this is necessary. It seems like generally this value is auto-calculated but it doesn't appear to be happening here without explicitly setting this.